File: lib-segment/src/main/java/websiteschema/mpsegment/pinyin/WordToPinyinClassfierFactory.java
package websiteschema.mpsegment.pinyin;

import websiteschema.mpsegment.conf.MPSegmentConfiguration;

import java.io.IOException;

public class WordToPinyinClassfierFactory {

    private final static WordToPinyinClassfierFactory instance = new WordToPinyinClassfierFactory();
    private final WordToPinyinClassifier classifier = new WordToPinyinClassifier();

    private WordToPinyinClassfierFactory() {
        try {
            WordToPinyinModel model = new WordToPinyinModel();
            model.load(MPSegmentConfiguration.getInstance().getPinyinModel());
            classifier.setModel(model);
        } catch (IOException ex) {
            ex.printStackTrace();
        }
    }

    public static WordToPinyinClassfierFactory getInstance() {
        return instance;
    }

    public WordToPinyinClassifier getClassifier() {
        return classifier;
    }
}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/pinyin/WordToPinyinClassfierFactory.scala
package websiteschema.mpsegment.pinyin

import websiteschema.mpsegment.conf.MPSegmentConfiguration
import websiteschema.mpsegment.hmm.{ViterbiImpl, HmmModel}

class WordToPinyinClassfierFactory {

  val model = new HmmModel(new ViterbiImpl)
  model.load(MPSegmentConfiguration().getPinyinModel())
  model.buildViterbi
  private val classifier = new WordToPinyinClassifier(model)

  def getClassifier(): WordToPinyinClassifier = {
    return classifier
  }
}

object WordToPinyinClassfierFactory {
  val instance = new WordToPinyinClassfierFactory()

  def apply() = instance
}
